orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/RestrictExecutionDuringTimeWindow.groovy

```diff
@@ -52,24 +58,20 @@ class RestrictExecutionDuringTimeWindow extends LinearStage {
     long backoffPeriod = 60000
     long timeout = TimeUnit.DAYS.toMillis(2)
 
-    private static final int HOUR_OF_DAY = Calendar.HOUR_OF_DAY
-    private static final int MINUTE = Calendar.MINUTE
-    private static final int SECOND = Calendar.SECOND
-
     private static final int DAY_START_HOUR = 0
     private static final int DAY_START_MIN = 0
     private static final int DAY_END_HOUR = 23
     private static final int DAY_END_MIN = 59
 
+    @Value('${tasks.executionWindow.timezone:America/Los_Angeles}')
+    String timeZoneId
+
     @Override
     TaskResult execute(Stage stage) {
-      TimeZone.'default' = TimeZone.getTimeZone("America/Los_Angeles")
-      System.setProperty("user.timezone", "America/Los_Angeles")
-
-      Date now = getCurrentDate()
+      Date now = new Date()
       Date scheduledTime
       try {
-        scheduledTime = getTimeInWindow(stage, getCurrentDate())
+        scheduledTime = getTimeInWindow(stage, new Date())
```

Comment:
The passed-in `scheduledTime` param gets mutated within `getTimeInWindow()`. I tried to keep the changes minimal, and it seems the in/out parameter is mainly to facilitate testing.